### ISSUE TITLE
linter: typescript/prefer-includes false positive on `/\d/.test(string)`

### ISSUE THREAD

**Issue by KieranP** [oxc-project/oxc#16001]
### What version of Oxlint are you using?

1.29.0

### What command did you run?

`oxlint --type-aware **/*.{js,ts,svelte}`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "ignorePatterns": ["/node_modules", "/*.{js,ts}"],
  "plugins": ["eslint", "oxc", "typescript", "import", "promise", "unicorn"],
  "categories": {
    "correctness": "error",
    "nursery": "error",
    "pedantic": "error",
    "perf": "error",
    "restriction": "error",
    "style": "error",
    "suspicious": "error"
  }
}
```

### What happened?

prefer-includes is reporting on `regex.test(string)`, but regexp is not a plain string value.

```
const numbers = /\d/
if (numbers.test(password)) {
  criteria = { ...criteria, numbers: true }
}
```

```
  × typescript-eslint(prefer-includes): Use `String#includes()` method with a string instead.
    ╭─[app/frontend/javascripts/helpers/validation.ts:35:7]
 34 │   const numbers = /\d/
 35 │   if (numbers.test(password)) {
    ·       ──────────────────────
 36 │     criteria = { ...criteria, numbers: true }
```

**Comment by camc314** [comment]
Please try `0.8.2`, it's releasing as I type this
